### PR TITLE
Profile Settings: UI

### PR DESCRIPTION
**Pull request by xx-TOOL-xx**
Profile setting form: complete UI and data collection logic

UPD: fixed according to comments;
conflicts solved

profileEditForm
profileEditForm2
profileEditForm3

### CODE REVIEW

File: src/components/UI/buttons/toggleButton/ToggleButton.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { IToggleButtonProps } from "components/UI/buttons/toggleButton/typesDef";
```
Comment (lyudad) [inline src/components/UI/buttons/toggleButton/ToggleButton.tsx]: ./typesDef

File: src/components/UI/datePicker/DatePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  return (
+    <Controller
+      name={`${type}[${index}].${formFieldName}`}
+      control={control}
+      defaultValue={null}
+      render={({ field }) => (
+        <DesktopDatePicker
+          label={label}
+          inputFormat="MM/dd/yyyy"
```
Comment (lyudad) [inline src/components/UI/datePicker/DatePicker.tsx]: move format to constants file

File: src/components/UI/datePicker/DatePicker.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+          value={field.value}
+          onChange={(date) => field.onChange(date)}
```
Comment (lyudad) [inline src/components/UI/datePicker/DatePicker.tsx]: onChange={field.onChange} should work

File: src/components/UI/select/Select.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { FormControl, MenuItem, Select as SelectM } from "@mui/material";
+import { ISelectProps } from "components/UI/select/typesDef";
+import { Controller } from "react-hook-form";
+import { SErrorMessage } from "components/UI/textField/style";
+
+function Select(props: ISelectProps) {
+  const { options, control, formFieldName, errors } = props;
+  const sxProps = {
```
Comment (lyudad) [inline src/components/UI/select/Select.tsx]: why we use it as variable here? why its not in styles?

File: src/components/UI/textField/TextField.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    helperText,
+    multiline,
+    rows,
+    maxLength,
+    formFieldName,
+    control,
+    errors,
+  } = props;
+  const endAdornment =
+    type === "number" ? (
```
Comment (lyudad) [inline src/components/UI/textField/TextField.tsx]: never compare with string always with constant

File: src/components/educationColumn/EducationColumn.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+  deleteButtonSx,
+} from "components/educationColumn/style";
+import { IEducationColumnProps } from "components/educationColumn/typesDef";
+import DatePicker from "components/UI/datePicker/DatePicker";
+import i18next from "localization/en/en.json";
+
+function EducationColumn(props: IEducationColumnProps) {
+  const { type, deletable, id, removeHandler, index, control } = props;
+  const inputsProps =
+    type === "education"
```
Comment (lyudad) [inline src/components/educationColumn/EducationColumn.tsx]: same, dont compare with string

File: src/components/educationColumn/EducationColumn.tsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+          formPositionFieldName: "jobTitle",
+        };
+
+  const handleRemoveButton = (currentId: string): void => {
+    removeHandler(currentId, index);
+  };
+
+  return (
+    <SWrapper>
+      {deletable ? (
```
Comment (lyudad) [inline src/components/educationColumn/EducationColumn.tsx]: {deletable && (
        <IconButton
          sx={deleteButtonSx}
          onClick={() => handleRemoveButton(id)}
          variant="outlined"
        >
          <DeleteOutlineIcon />
        </IconButton>
      )} should work

File: src/components/layoutElementWithTitle/LayoutElementWithTitle.tsx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    />
+  );
+
+  const renderSkillsLayout = () =>
+    skillsOptions ? (
+      <SkillsLayout errors={errors} control={control} options={skillsOptions} />
+    ) : null;
+
+  const renderElement = (type: typeof element): JSX.Element | null => {
+    switch (type) {
```
Comment (lyudad) [inline src/components/layoutElementWithTitle/LayoutElementWithTitle.tsx:112]: interesting solution, but again, no strings, it should be constant, which you will use everywhere for compare

File: src/components/UI/buttons/toggleButton/ToggleButton.tsx (hunk 1 above)
Comment (xx-TOOL-xx) [reply src/components/UI/buttons/toggleButton/ToggleButton.tsx]: Screenshot from 2022-08-15 15-19-12
sorry, as I understand this comment about non absolute import but in code this import is absolute

File: src/components/UI/datePicker/DatePicker.tsx (hunk 2 above)
Comment (xx-TOOL-xx) [reply src/components/UI/datePicker/DatePicker.tsx]: fixed

File: src/components/UI/datePicker/DatePicker.tsx (hunk 3 above)
Comment (xx-TOOL-xx) [reply src/components/UI/datePicker/DatePicker.tsx]: fixed

File: src/components/UI/select/Select.tsx (hunk 4 above)
Comment (xx-TOOL-xx) [reply src/components/UI/select/Select.tsx]: fixed

File: src/components/UI/textField/TextField.tsx (hunk 5 above)
Comment (xx-TOOL-xx) [reply src/components/UI/textField/TextField.tsx]: fixed

File: src/components/educationColumn/EducationColumn.tsx (hunk 6 above)
Comment (xx-TOOL-xx) [reply src/components/educationColumn/EducationColumn.tsx]: fixed

File: src/components/educationColumn/EducationColumn.tsx (hunk 7 above)
Comment (xx-TOOL-xx) [reply src/components/educationColumn/EducationColumn.tsx]: fixed

File: src/components/layoutElementWithTitle/LayoutElementWithTitle.tsx (hunk 8 above)
Comment (xx-TOOL-xx) [reply src/components/layoutElementWithTitle/LayoutElementWithTitle.tsx:112]: fixed

**Pull request merged**